react/features/virtual-background/components/VirtualBackgroundPreview.js

```diff
@@ -0,0 +1,268 @@
+     * Apply background effect on video preview.
+     *
+     * @returns {Promise}
+     */
+    async _applyBackgroundEffect() {
+        const jitsiTrack = this.state.trackData[0].jitsiTrack;
+
+        this.setState({ loading: true });
+        await this.props.dispatch(toggleBackgroundEffect(this.props.options, jitsiTrack));
+        this.setState({ loading: false });
+    }
+
+    /**
+     * Renders a preview entry.
+     *
+     * @param {Object} data - The track data.
+     * @param {number} index - The index of the entry.
+     * @returns {React$Node}
+     */
+    _renderPreviewEntry(data, index) {
+        const { error, jitsiTrack } = data;
+        const { t } = this.props;
+        const key = `vp-${index}`;
+        const className = 'video-background-preview-entry';
+
+        if (this.state.loading) {
+            return (
+                <div>
+                    <span className = 'loading-content-text'>{t('virtualBackground.pleaseWait')}</span>
+                    <Spinner
+                        invertColor = { true }
+                        isCompleting = { false }
+                        size = { 'large' } />
+                </div>
+            );
+        }
+        if (error) {
+            return (
+                <div
+                    className = { className }
+                    key = { key }
+                    video-preview-container = { true }>
+                    <div className = 'video-preview-error'>{t(error)}</div>
+                </div>
+            );
+        }
+
+        const props: Object = {
+            className,
+            key
+        };
+
+        return (
+            <div { ...props }>
+                <Video
+                    className = { videoClassName }
+                    playsinline = { true }
+                    videoTrack = {{ jitsiTrack }} />
+            </div>
+        );
+    }
+
+    /**
+     * Implements React's {@link Component#componentDidMount}.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this._setTracks();
+    }
+
+    /**
+     * Implements React's {@link Component#componentWillUnmount}.
+     *
+     * @inheritdoc
+     */
+    componentWillUnmount() {
+        this._componentWasUnmounted = true;
+        this._disposeTracks(this.state.trackData);
+    }
+
+    /**
+     * Implements React's {@link Component#componentDidUpdate}.
+     *
+     * @inheritdoc
+     */
+    async componentDidUpdate(prevProps) {
+        if (!equals(this.props.videoDeviceIds, prevProps.videoDeviceIds)) {
+            this._setTracks();
+        }
+        if (!equals(this.props.options, prevProps.options)) {
+            this._applyBackgroundEffect();
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { trackData } = this.state;
+
+        return <div className = 'video-preview'>{trackData.map((data, i) => this._renderPreviewEntry(data, i))}</div>;
```

Comment:
Are we showing a preview for each camera here? We should have a selector for the cameras instead I think, similar to the preview in the settings. Can you ask Tibi about this?